chalicelib_smaht/checks/helpers/google_utils.py

```diff
@@ -0,0 +1,1107 @@
+            #     { 'name': 'itemName' },
+            #     { 'name': 'itemId' },
+            #     { 'name': 'itemCategory2' },
+            #     { 'name': 'itemBrand' }
+            # ]
+            report_request_json["dimensions"] = [
+                { 'name': 'customEvent:name' },
+                { 'name': 'customEvent:lab' },
+                { 'name': 'customEvent:file_classification' },
+                { 'name': 'customEvent:file_type' }
+            ]
+            report_request_json["limit"] = 100
+            if execute:
+                return self.query_reports([report_request_json])
+            return report_request_json
+
+        ##############################################
+        #### Metadata.tsv File Download Analytics ####
+        ##############################################
+
+
+        @report
+        def metadata_tsv_by_country(self, start_date='yesterday', end_date='yesterday', execute=True):
+            report_request_json = {
+                'date_ranges' : [{ 'start_date' : start_date, 'end_date' : end_date }],
+                'metrics': [
+                    { 'name': 'itemsCheckedOut' },
+                ],
+                'dimensions': [
+                    { 'name': 'country' },
+                    { 'name': 'itemListName' }
+                ],
+                "order_bys" : [
+                    { 'metric' : { 'metric_name': 'itemsCheckedOut' }, 'desc': True },
+                ],
+                'dimension_filter' : {
+                    'and_group': {
+                        'expressions': [
+                            {
+                                "filter" : { 
+                                    'field_name' : "customEvent:source", 
+                                    'string_filter': { 
+                                        "value" : "SelectedFilesDownloadModal", 
+                                        "match_type" : "EXACT" 
+                                    } 
+                                }
+                        }]
+                    }
+                },
+                'limit' : 100
+            }
+            if execute:
+                return self.query_reports([report_request_json])
+            return report_request_json
+
+
+
+
+    class SheetsAPI(_NestedGoogleServiceAPI):
+        '''
+        Use this sub-class to help query, read, edit, and analyze spreadsheet data, which will be returned in form of multi-dimensional JSON array.
+
+        TODO: Implement
+        '''
+        pass
+
+
+
+
+    class DocsAPI(_NestedGoogleServiceAPI):
+        '''
+        Use this sub-class to help query, read, and edit Google Doc data.
+
+        TODO: Implement
+        '''
+        pass
+
+
+
+
+######### CODE TO TEST THE ABOVE #########
+
+'''
+The following is code to test the above class(es).
+Run this file in interactive mode and continue on:
+
+> python3 -i google_utils.py
+
+'''
+
+
+if __name__ == "__main__":
```

Comment:
When you're done testing consider trying to write unit tests in utils to mimic what you're doing interactively